internal/diagnostics/logs.go

```diff
@@ -0,0 +1,184 @@
+	}
+}
+
+// getPodsForModule gets all pods for a specific module in kyma-system namespace
+func (c *ModuleLogsCollector) getPodsForModule(ctx context.Context, moduleName string) ([]corev1.Pod, error) {
+	labelSelector := map[string]string{
+		"kyma-project.io/module": moduleName,
+	}
+
+	deploymentsList, err := c.client.Static().AppsV1().Deployments("kyma-system").List(ctx, metav1.ListOptions{
+		LabelSelector: resources.LabelSelectorFor(labelSelector),
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to list deployments for module %s: %w", moduleName, err)
+	}
+
+	podsList := make([]corev1.Pod, 0)
+
+	for _, deployment := range deploymentsList.Items {
+		matchLabels := deployment.Spec.Selector.MatchLabels
+		pods, err := c.client.Static().CoreV1().Pods("kyma-system").List(ctx, metav1.ListOptions{
+			LabelSelector: resources.LabelSelectorFor(matchLabels),
+		})
+		if err != nil {
+			return nil, fmt.Errorf("failed to list pods for deployment %s: %w", deployment.Name, err)
+		}
+
+		podsList = append(podsList, pods.Items...)
+	}
+
+	return podsList, nil
+}
+
+// collectPodLogs collects error logs from a pod
+func (c *ModuleLogsCollector) collectPodLogs(ctx context.Context, pod *corev1.Pod, options LogOptions, allLogs map[string][]string) {
+	containers := append(pod.Spec.Containers, pod.Spec.InitContainers...)
+
+	for _, container := range containers {
+		containerLogs := c.collectContainerLogs(ctx, pod, container.Name, options)
+		key := fmt.Sprintf("%s/%s", pod.Name, container.Name)
+		allLogs[key] = append(allLogs[key], containerLogs...)
+	}
+}
+
+// collectContainerLogs collects and filters error logs from a container
+func (c *ModuleLogsCollector) collectContainerLogs(ctx context.Context, pod *corev1.Pod, containerName string, options LogOptions) []string {
+	logOptions := c.buildPodLogOptions(containerName, options)
+
+	req := c.client.Static().CoreV1().Pods(pod.Namespace).GetLogs(pod.Name, logOptions)
+
+	logStream, err := req.Stream(ctx)
+	if err != nil {
+		out.Verbosefln("Failed to get log stream for container %s in pod %s: %v", containerName, pod.Name, err)
+		return []string{}
+	}
+	defer logStream.Close()
+
+	// Attempt structured JSON parsing; fallback to error token filtering
+	return c.extractStructuredOrErrorLogs(logStream, pod.Name, containerName)
+}
+
+// extractStructuredOrErrorLogs parses each log line strictly as JSON and only keeps lines
+// that contain the keys: level, msg, and ts. Non-JSON lines or JSON without these keys are ignored.
+// Only entries whose level is one of error, fatal, panic are returned.
+func (c *ModuleLogsCollector) extractStructuredOrErrorLogs(logStream io.ReadCloser, podName, containerName string) []string {
+	scanner := bufio.NewScanner(logStream)
+	var collected []string
+
+	for scanner.Scan() {
+		line := scanner.Text()
+		var obj map[string]any
+		if err := json.Unmarshal([]byte(line), &obj); err != nil {
+			// Not JSON -> ignore
+			continue
```

Comment:
Maybe we should add a note at the end of the log saying "some lines were skipped because of the wrong format."